apis.md

```diff
@@ -9,3 +9,10 @@ The following OpenTracing APIs are currently available in the RFC stage:
 
 Please refer to the README files in the respective projects for examples of usage.
 
+### Implementation Notes
+
+OpenTracing APIs intend to express a similar experience to the user, while still being idiomatic to the language or framework. The following concepts apply to all variants.
+
+A **Span** is shared mutable state. It accumulates tags and logs, and can also create children. Finishing a span means capturing its state as a SpanRecord.
+
+A **SpanRecord** is an immutable representation of a completed span. It can be reported directly to an OpenTracing system, or encoded into a representation of an alternate system, such as HTrace or Zipkin.
```

Comment:
>
> @adriancole <https://github.com/adriancole> Can you elaborate which
> features you would want to see in such Mock tracer? And if we have
> example/zipkin-like converted to a real reference implementation, wouldn't
> it be sufficient as a playground? I am a bit uneasy about maintaining a
> full-featured "fake" implementation.
>
Fake/InMemory/Buffering tracer would have no dependencies. Ex many tracing
systems require that you start processes etc. Fake could keep state
in-memory which would allow us to define tests. This approach is quite
common. In fact, zipkin itself, both java and scala, have a in-memory span
store for the purpose of defining tests. Finagle has a BufferingTracer,
etc. This is also quite common outside tracing. For example, most large
projects I've worked on, jclouds, okhttp, etc. have a fake. Also, internal
work at square, we used fakes as ways of dealing with microservice
architecture testing when a boundary isn't in your teams control. Long
story short, is that we can make this sort of rigor optional, but I'd
certainly not write a system without a fake in 2016.

For users, I suspect a LoggingTracer would be the most useful way to
step-into the project. This again has no dependencies, so they don't need
to run docker or any other external process. Again, this is extremely
normal behavior, and for example, finagle has a console tracer.